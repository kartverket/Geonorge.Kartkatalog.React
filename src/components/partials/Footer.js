// Dependencies
import React, { Component } from 'react';
import { connect } from 'react-redux';

// Actions
import { getResource } from 'actions/ResourceActions'
import { getGeonorgeLogo } from 'actions/ImageActions';

// Stylesheets
import style from 'components/partials/Footer.module.scss';

// Assets
import kartverketLogo from 'images/kartverket.png';


export class Footer extends Component {

	getBuildNumber() {
		return this.props.environment && this.props.environment.buildNumber
		? (
				<p>
					<small>{this.props.getResource('Version', 'Versjon')} {this.props.environment.buildNumber}</small>
				</p>
		)
		: '';
	}

	render() {
		const isMapRoute = this.props.router && this.props.router.location && this.props.router.location.pathname === "/kart";
		return !isMapRoute ? (
			<div className={style.container}>
				<div className={style.logo}>
					<img src={this.props.getGeonorgeLogo()} alt="Geonorge logo" title="Logo for Geonorge" />
					{this.getBuildNumber()}
				</div>
				<div className={style.about}>
					<h2 className={style.heading}>{this.props.getResource('AboutTheSite', 'Om nettstedet')}</h2>
					<p>
						<a href="https://www.geonorge.no/aktuelt/om-geonorge/">
							{this.props.getResource('AboutGeonorge', 'Om Geonorge')}
						</a>
					</p>
					<p>
						<a href="https://www.geonorge.no/aktuelt/Nyheter/annet/personvern-og-bruk-av-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570">
							{this.props.getResource('PrivacyAndUseOfCookies', 'Personvern og bruk av cookies')}
						</a>
					</p>
				</div>
				<div className={style.contact}>
					<h2 className={style.heading}>{this.props.getResource('ContactUs', 'Kontakt')}</h2>
					<p>{this.props.getResource('Telephone', 'Telefon')}: 32 11 80 00</p>
					<p>post@norgedigitalt.no</p>
					<p>{this.props.getResource('OrgNo', 'Org. nr.')}: 971 040 238</p>
					<hr />
					{this.props.getResource('ASolutionBy', 'En l√∏sning fra')}
					<div className={style.logo}><img src={kartverketLogo} alt="Kartverket logo" title="Logo for Kartverket" /></div>
				</div>
			</div>
		) : ''
	}
}


const mapStateToProps = state => ({
    router: state.router,
	resources: state.resources,
	environment: state.environment
});

const mapDispatchToProps = {
	getResource,
    getGeonorgeLogo
};

export default connect(mapStateToProps, mapDispatchToProps)(Footer);
