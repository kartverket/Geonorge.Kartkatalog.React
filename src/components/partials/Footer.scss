@import '../../scss/abstracts/all';

:local(.container) {
    display: flex;
    flex-direction: column;
    @include breakpoint(phablet) {
        flex-direction: row;
    }
   
    margin: 0 auto;
    justify-content: space-between;    
    margin-bottom: 40px;
    > div {        
        margin: 10px;
        &:local(.logo) {
            width: 33%;
            img {
                max-width: 256px;
                padding-top: 30px;
            }

        }
        &:local(.about) {
            width: 33%;
        }
        &:local(.contact) {
            width: 33%;
            hr {
                border-color: #333;
            }
        }
    }
}