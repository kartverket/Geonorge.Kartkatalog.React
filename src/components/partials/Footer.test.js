import React from 'react';
import {shallow} from "enzyme";
import {Footer} from './Footer';

import style from './Footer.scss';

function setup() {
    const wrapper = shallow(<Footer/>);
    return {
        wrapper
    }
}

describe('Footer', () => {
    it('should render self and subcomponents', () => {
        const {wrapper} = setup();
        expect(wrapper.hasClass(style.container)).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('should include 2 images', () => {
        const {wrapper} = setup();
        expect(wrapper.find('img').length).toBe(2);
    })
});
