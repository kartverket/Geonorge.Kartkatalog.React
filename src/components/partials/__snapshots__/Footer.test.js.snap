// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render self and subcomponents 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        >
          <img
            alt="logo geonorge"
            src="test-file-stub"
            title="Logo for Geonorge"
          />
        </div>,
        <div
          className="about"
        >
          <h3>
            OM NETTSTEDET
          </h3>
          <p>
            <a
              href="https://www.geonorge.no/aktuelt/om-geonorge/"
            >
              Om Geonorge
            </a>
          </p>
          <p>
            <a
              href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
            >
              Personvern og bruk av cookies
            </a>
          </p>
        </div>,
        <div
          className="contact"
        >
          <h3>
            KONTAKT
          </h3>
          <p>
            Telefon: 32 11 80 00
          </p>
          <p>
            post@norgedigitalt.no
          </p>
          <p>
            Org. nr.: 971 040 238
          </p>
          <hr />
          En løsning fra
          <div
            className="logo"
          >
            <img
              alt="logo geonorge"
              src="test-file-stub"
              title="Logo for Kartverket"
            />
          </div>
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo geonorge"
            src="test-file-stub"
            title="Logo for Geonorge"
          />,
          "className": "logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo geonorge",
            "src": "test-file-stub",
            "title": "Logo for Geonorge",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              OM NETTSTEDET
            </h3>,
            <p>
              <a
                href="https://www.geonorge.no/aktuelt/om-geonorge/"
              >
                Om Geonorge
              </a>
            </p>,
            <p>
              <a
                href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
              >
                Personvern og bruk av cookies
              </a>
            </p>,
          ],
          "className": "about",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OM NETTSTEDET",
            },
            "ref": null,
            "rendered": "OM NETTSTEDET",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="https://www.geonorge.no/aktuelt/om-geonorge/"
              >
                Om Geonorge
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Om Geonorge",
                "href": "https://www.geonorge.no/aktuelt/om-geonorge/",
              },
              "ref": null,
              "rendered": "Om Geonorge",
              "type": "a",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
              >
                Personvern og bruk av cookies
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Personvern og bruk av cookies",
                "href": "http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570",
              },
              "ref": null,
              "rendered": "Personvern og bruk av cookies",
              "type": "a",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              KONTAKT
            </h3>,
            <p>
              Telefon: 32 11 80 00
            </p>,
            <p>
              post@norgedigitalt.no
            </p>,
            <p>
              Org. nr.: 971 040 238
            </p>,
            <hr />,
            "En løsning fra",
            <div
              className="logo"
            >
              <img
                alt="logo geonorge"
                src="test-file-stub"
                title="Logo for Kartverket"
              />
            </div>,
          ],
          "className": "contact",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "KONTAKT",
            },
            "ref": null,
            "rendered": "KONTAKT",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Telefon: 32 11 80 00",
            },
            "ref": null,
            "rendered": "Telefon: 32 11 80 00",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "post@norgedigitalt.no",
            },
            "ref": null,
            "rendered": "post@norgedigitalt.no",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Org. nr.: 971 040 238",
            },
            "ref": null,
            "rendered": "Org. nr.: 971 040 238",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          "En løsning fra",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="logo geonorge"
                src="test-file-stub"
                title="Logo for Kartverket"
              />,
              "className": "logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo geonorge",
                "src": "test-file-stub",
                "title": "Logo for Kartverket",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          >
            <img
              alt="logo geonorge"
              src="test-file-stub"
              title="Logo for Geonorge"
            />
          </div>,
          <div
            className="about"
          >
            <h3>
              OM NETTSTEDET
            </h3>
            <p>
              <a
                href="https://www.geonorge.no/aktuelt/om-geonorge/"
              >
                Om Geonorge
              </a>
            </p>
            <p>
              <a
                href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
              >
                Personvern og bruk av cookies
              </a>
            </p>
          </div>,
          <div
            className="contact"
          >
            <h3>
              KONTAKT
            </h3>
            <p>
              Telefon: 32 11 80 00
            </p>
            <p>
              post@norgedigitalt.no
            </p>
            <p>
              Org. nr.: 971 040 238
            </p>
            <hr />
            En løsning fra
            <div
              className="logo"
            >
              <img
                alt="logo geonorge"
                src="test-file-stub"
                title="Logo for Kartverket"
              />
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo geonorge"
              src="test-file-stub"
              title="Logo for Geonorge"
            />,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo geonorge",
              "src": "test-file-stub",
              "title": "Logo for Geonorge",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                OM NETTSTEDET
              </h3>,
              <p>
                <a
                  href="https://www.geonorge.no/aktuelt/om-geonorge/"
                >
                  Om Geonorge
                </a>
              </p>,
              <p>
                <a
                  href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
                >
                  Personvern og bruk av cookies
                </a>
              </p>,
            ],
            "className": "about",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OM NETTSTEDET",
              },
              "ref": null,
              "rendered": "OM NETTSTEDET",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://www.geonorge.no/aktuelt/om-geonorge/"
                >
                  Om Geonorge
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Om Geonorge",
                  "href": "https://www.geonorge.no/aktuelt/om-geonorge/",
                },
                "ref": null,
                "rendered": "Om Geonorge",
                "type": "a",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570"
                >
                  Personvern og bruk av cookies
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Personvern og bruk av cookies",
                  "href": "http://kartverket.no/Om-Kartverket/Personvern-og-cookies/#_ga=2.93024157.1065069024.1548754965-263923791.1536837570",
                },
                "ref": null,
                "rendered": "Personvern og bruk av cookies",
                "type": "a",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                KONTAKT
              </h3>,
              <p>
                Telefon: 32 11 80 00
              </p>,
              <p>
                post@norgedigitalt.no
              </p>,
              <p>
                Org. nr.: 971 040 238
              </p>,
              <hr />,
              "En løsning fra",
              <div
                className="logo"
              >
                <img
                  alt="logo geonorge"
                  src="test-file-stub"
                  title="Logo for Kartverket"
                />
              </div>,
            ],
            "className": "contact",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "KONTAKT",
              },
              "ref": null,
              "rendered": "KONTAKT",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Telefon: 32 11 80 00",
              },
              "ref": null,
              "rendered": "Telefon: 32 11 80 00",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "post@norgedigitalt.no",
              },
              "ref": null,
              "rendered": "post@norgedigitalt.no",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Org. nr.: 971 040 238",
              },
              "ref": null,
              "rendered": "Org. nr.: 971 040 238",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            "En løsning fra",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="logo geonorge"
                  src="test-file-stub"
                  title="Logo for Kartverket"
                />,
                "className": "logo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo geonorge",
                  "src": "test-file-stub",
                  "title": "Logo for Kartverket",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
